vpr/src/route/rr_graph2.cpp

```diff
@@ -2628,19 +2448,19 @@ static void label_incoming_wires(const int chan_num,
     *num_ending_wires = num_ending;
 }
 
-static int find_label_of_track(int* wire_mux_on_track,
+static int find_label_of_track(const std::vector<RRNodeId>& wire_mux_on_track,
                                int num_wire_muxes,
                                int from_track) {
     /* Returns the index/label in array wire_mux_on_track whose entry equals from_track. If none are
      * found, then returns the index of the entry whose value is the largest */
     int i_label = -1;
-    int max_track = -1;
+    RRNodeId max_track = RRNodeId::INVALID();
 
     for (int i = 0; i < num_wire_muxes; i++) {
-        if (wire_mux_on_track[i] == from_track) {
```

Comment:
It seems like label is a track number here (from 0 to W-1) so this implies "labels" are really "track numbers" and should be kept as ints, I think.